Sample/Kitchen Sink/DemonstrationTools/UnitConverter.swift

```diff
@@ -0,0 +1,184 @@
+        MenuField(selection: $demonstrationVM.boundSelectionInput,
+                  options: UnitConverterViewModel.LengthType.allCases,
+                  placeholder: "")
+        .menuFieldStyle(.default)
+        .padding([.bottom], themeManager.current.padding.grid2x)
+    }
+
+    var lengthOutputUnitPicker: some View {
+        MenuField(selection: $demonstrationVM.boundSelectionOutput,
+                  options: UnitConverterViewModel.LengthType.allCases,
+                  placeholder: "")
+        .menuFieldStyle(.default)
+        .padding([.bottom], themeManager.current.padding.grid2x)
+    }
+
+    var kgInputField: some View {
+        InputField(text: $demonstrationVM.kgInputString,
+                   placeholder: demonstrationVM.kgHint,
+                   bottomLabelConfig: BottomLabelConfig(demonstrationVM.bottomKgLabel),
+                   formatter: { typedString in
+            guard let decimalValue = Decimal(string: typedString) else { return typedString }
+            return demonstrationVM.toString2DP(value: decimalValue)
+        }, filter: .doubleOnly)
+        .textFieldStyle(demonstrationVM.emptyFields ? DefaultTextFieldStyle.caution : DefaultTextFieldStyle.default)
+    }
+
+    var lbInputField: some View {
+        InputField(text: $demonstrationVM.lbsInputString,
+                   placeholder: demonstrationVM.lbHint,
+                   bottomLabelConfig: BottomLabelConfig(demonstrationVM.bottomlbLabel),
+                   formatter: { typedString in
+            guard let decimalValue = Decimal(string: typedString) else { return typedString }
+            return demonstrationVM.toString2DP(value: decimalValue)
+        }, filter: .doubleOnly)
+        .textFieldStyle(demonstrationVM.emptyFields ? DefaultTextFieldStyle.caution : DefaultTextFieldStyle.default)
+    }
+
+    var swapButton: some View {
+        Button(action: {
+            withAnimation(.easeInOut(duration: 0.5)) {
+                swapFields()
+            }
+        }, label: {
+            Image(systemName: "arrow.left.arrow.right")
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+                .frame(width: 40, height: 40)
+                .foregroundColor(theme.color.nominal)
+                .padding([.bottom], theme.padding.grid2x)
+        })
+    }
+
+    var convertButton1: some View {
+        Button(action: {
+            convertStaticUnits()
+        }, label: {
+            Text(demonstrationVM.convert)
+                .padding([.bottom], theme.padding.grid2x)
+        })
+        .buttonStyle(.text)
+    }
+
+    var convertButton2: some View {
+        Button(action: {
+            runLengthConversion()
+        }, label: {
+            Text(demonstrationVM.convert)
+                .padding([.bottom], theme.padding.grid2x)
+        })
+        .buttonStyle(.text)
+    }
+
+    func convertStaticUnits() {
+        if !demonstrationVM.fieldsAreEmpty() {
```

Comment:
Minor, these don't need to be functions inside the view, can all be in the viewModel, also better for the check on if fields are empty to be in the view model function as well - allows for testing, reuse of function, and error handling 👍